src/win/winpty.cc

```diff
@@ -276,20 +292,26 @@ static NAN_METHOD(PtyKill) {
   if (info.Length() != 2 ||
       !info[0]->IsNumber() ||
       !info[1]->IsNumber()) {
-    Nan::ThrowError("Usage: pty.kill(pid, innerPidHandle)");
+    Nan::ThrowError("Usage: pty.kill(pid, innerPid)");
     return;
   }
 
-  HANDLE handle = reinterpret_cast<HANDLE>(static_cast<int64_t>(info[0]->Int32Value(Nan::GetCurrentContext()).FromJust()));
-  HANDLE innerPidHandle = reinterpret_cast<HANDLE>(static_cast<int64_t>(info[1]->Int32Value(Nan::GetCurrentContext()).FromJust()));
+  DWORD pid = info[0]->Uint32Value(Nan::GetCurrentContext()).FromJust();
+  DWORD innerPid = info[1]->Uint32Value(Nan::GetCurrentContext()).FromJust();
 
-  winpty_t *pc = get_pipe_handle(handle);
+  winpty_t *pc = get_pipe_handle(pid);
   if (pc == nullptr) {
     Nan::ThrowError("Pty seems to have been killed already");
     return;
   }
 
-  assert(remove_pipe_handle(handle));
+  assert(remove_pipe_handle(pid));
+
+  HANDLE innerPidHandle = OpenProcess(PROCESS_TERMINATE, FALSE, innerPid);
+  if (innerPidHandle == nullptr) {
+    return info.GetReturnValue().SetUndefined();
+  }
+  TerminateProcess(innerPidHandle, 0);
```

Comment:
According to [closehandle#remarks](https://learn.microsoft.com/en-us/windows/win32/api/handleapi/nf-handleapi-closehandle#remarks), "Closing a process handle does not terminate the associated process or remove the process object." Therefore, I added a `TerminateProcess` call here to ensure the process would be terminated, but I'm not sure whether it was previously being terminated elsewhere?
CC @Tyriar

Reply:
the termination happens here https://github.com/microsoft/node-pty/blob/d01f07df6acb4b0629a74f7a61423d4b87c81f91/src/windowsPtyAgent.ts#L178-L192, if `TerminateProcess` can guarantee the exit of all process in the console tree then I am good with the change. This should be a follow-up to avoid breaking behaviors. I suggest to revert the change in this PR.

Reply:
I reverted the change.